bugsnag-android-core/src/main/java/com/bugsnag/android/BugsnagJournal.kt

```diff
@@ -16,27 +17,33 @@ import kotlin.concurrent.thread
  * The single instance to be used in Bugsnag is "instance", and startInstance() must be called before
  * using it.
  */
-internal class BugsnagMainJournal internal constructor(
+internal class BugsnagJournal @JvmOverloads internal constructor (
     private val logger: Logger,
     baseDocumentPath: File,
-    initialDocument: Map<String, Any>
-) {
+    initialDocument: Map<String, Any> = mapOf()
+) : Closeable {
     internal val journal = JournaledDocument(
         baseDocumentPath,
         journalType,
         journalVersion,
         mmapBufferSize,
         highWater,
-        initialDocument
+        withInitialDocumentContents(initialDocument)
     )
 
+    init {
+        beginSnapshotHousekeepingThread()
+    }
+
     /**
      * The document in its current state (with all current journal entries applied).
      * Do NOT modify any of the document's contents directly; information will be lost if you do.
      * Use addCommand() instead to modify the document.
      */
     val document: Map<String, Any> get() = journal
 
+    private var housekeepingThreadShouldKeepRunning = false
```

Comment:
Does this need to be marked as `@Volatile` if it could be set/read from multiple threads?

Reply:
It's never read by another thread, and only changed when the journal is closed. The journal's housekeeping thread runs until this value becomes false (which only happens when close is called).